skrifa/src/outline/glyf/mod.rs

```diff
@@ -276,9 +248,116 @@ impl<'a> Outlines<'a> {
     }
 }
 
-struct Scaler<'a> {
+trait Scaler {
+    fn coords(&self) -> &[F2Dot14];
+    fn outlines(&self) -> &Outlines;
+    fn setup_phantom_points(
+        &mut self,
+        bounds: [i16; 4],
+        lsb: i32,
+        advance: i32,
+        tsb: i32,
+        vadvance: i32,
+    );
+    fn load_simple(&mut self, glyph: &SimpleGlyph, glyph_id: GlyphId) -> Result<(), DrawError>;
+    fn load_composite(
+        &mut self,
+        glyph: &CompositeGlyph,
+        glyph_id: GlyphId,
+        recurse_depth: usize,
+    ) -> Result<(), DrawError>;
+
+    fn load(
+        &mut self,
+        glyph: &Option<Glyph>,
+        glyph_id: GlyphId,
+        recurse_depth: usize,
+    ) -> Result<(), DrawError> {
+        if recurse_depth > GLYF_COMPOSITE_RECURSION_LIMIT {
+            return Err(DrawError::RecursionLimitExceeded(glyph_id));
+        }
+        let glyph = match &glyph {
+            Some(glyph) => glyph,
+            // This is a valid empty glyph
+            None => return Ok(()),
+        };
+        let bounds = [glyph.x_min(), glyph.x_max(), glyph.y_min(), glyph.y_max()];
+        let outlines = self.outlines();
+        let coords: &[F2Dot14] = self.coords();
+        let lsb = outlines.lsb(glyph_id, coords);
+        let advance = outlines.advance_width(glyph_id, coords);
```

Comment:
HB diff: these have variations applied... HB always uses base values and computes phantom point variations with gvar

Reply:
Deferred to https://github.com/googlefonts/fontc/issues/784